src/Collection/CollectionTrait.php

```diff
@@ -1110,6 +1110,71 @@ public function countKeys(): int
         return count($this->toArray());
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function keys(): CollectionInterface
+    {
+        $generator = function (): Generator {
+            foreach ($this->optimizeUnwrap() as $key => $value) {
+                yield $key;
+            }
+        };
+
+        return $this->newCollection($generator());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function values(): CollectionInterface
+    {
+        $generator = function (): Generator {
+            foreach ($this->optimizeUnwrap() as $value) {
+                yield $value;
+            }
+        };
+
+        return $this->newCollection($generator());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function implode(string $glue, callable|string|null $path = null): string
+    {
+        $items = $this;
+        if ($path !== null) {
+            $items = $items->extract($path);
+        }
+
+        return implode($glue, $items->toList());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function when(mixed $condition, callable $callback): CollectionInterface
+    {
+        if ($condition) {
+            return $callback($this, $condition);
+        }
```

Comment:
I'm not sure how this helps exactly. It's just an if statement to call the same code that's passed in.